### PR TITLE
Enhancement / expression function distance

### PR DESCRIPTION
**Pull request by SteRiccio**
## Related Issue

Resolves #CU-fv4n9e

## Types of changes

- [ ] Docs change
- [ ] Refactoring
- [ ] Dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Brief description of the changes proposed and/or how the issue(s) has(have) been solved.

Added 'distance' function to custom record expression functions.
Improved test SurveyBuilder adding category builder.

## How has this been tested

Unit tests have been added, testing the function with constant values or evaluated node values as arguments.

##### Do you consider this PR needs further testing?

- [x] No
- [ ] Yes

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-fv4n9e ExpressionFunction: distance](https://app.clickup.com/t/fv4n9e)

File: core/geo/geoUtils.js [hunk 1]
```diff
@@ -51,23 +56,68 @@ export const findSrsByCodeOrName = (codeOrName, limit = 200) =>
 
 const getSrsByCode = (code) => srsByCode[code]
 
+const toLatLon = (point) => {
```
Comment (minotogna) [inline core/geo/geoUtils.js:59]: toLatLng (by convention it's use Lng)

File: core/geo/geoUtils.js [hunk 2]
```diff
@@ -51,23 +56,68 @@ export const findSrsByCodeOrName = (codeOrName, limit = 200) =>
...
+    // projection is not needed
+    return point
+  }
+  const x = Point.getX(point)
+  const y = Point.getY(point)
+
+  const srsFrom = getSrsByCode(srsId)
+  const srsTo = Srs.latLonSrs
+
+  const lonLat = proj4(
```
Comment (minotogna) [inline core/geo/geoUtils.js:73]: latLng

File: core/geo/geoUtils.js [hunk 3]
```diff
@@ -51,23 +56,68 @@ export const findSrsByCodeOrName = (codeOrName, limit = 200) =>
...
-    : proj4(
-        Srs.getWkt(srs), // From srs
-        Srs.getWkt(Srs.latLonSrs), // To lat lon
-        [x, y] // Coordinates
-      )
+  const earthRadius = 6371000 // Earth radius in meters
+  const dLat = toRad(lat2 - lat1)
+  const dLon = toRad(lon2 - lon1)
+  const lat1Rad = toRad(lat1)
+  const lat2Rad = toRad(lat2)
```
Comment (minotogna) [inline core/geo/geoUtils.js:118]: use Lng

**Review by minotogna — CHANGES_REQUESTED** [review]

File: core/geo/geoUtils.js (hunk 1 above)
Comment (SteRiccio) [reply core/geo/geoUtils.js:59]: lat lon or lat long is more common, I vote for "lat long"

File: core/geo/geoUtils.js (hunk 2 above)
Comment (SteRiccio) [reply core/geo/geoUtils.js:73]: or better, [lon lat]; proj4 returns an array with x (longitude)  and y (latitude) coordinates)

File: core/geo/geoUtils.js (hunk 3 above)
Comment (SteRiccio) [reply core/geo/geoUtils.js:118]: ok

**Review by minotogna — APPROVED** [review]

**Review by ramirobg94 — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**